src/main/java/lotto/domain/RankStatistic.java

```diff
@@ -0,0 +1,53 @@
+package lotto.domain;
+
+import static lotto.domain.LottoGames.LOTTO_PRICE;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import lotto.domain.model.Rank;
+
+public class RankStatistic {
+
+  private Map<Rank, Long> rankStatistic;
+
+  public RankStatistic(List<Rank> ranks) {
+    rankStatistic = ranks.stream()
+        .collect(
+            Collectors.groupingBy(
+                Function.identity(),
+                Collectors.counting()
+            )
+        );
+  }
+
+  public double calculateYield() {
+    return calculatePrizeMoney() / (double) calculatePurchaseAmount();
+  }
+
+  private long calculatePrizeMoney() {
+    return rankStatistic.entrySet()
+        .stream()
+        .mapToLong(entry -> {
+          Rank rank = entry.getKey();
+          long winCount = entry.getValue();
+          return rank.getWinningMoney() * winCount;
+        })
+        .sum();
+  }
```

Comment:
```suggestion
    private long sumTotalPrizeMoney() {
        return rankStatistic.entrySet()
                .stream()
                .mapToLong(this::calculatePrizeMoney)
                .sum();
    }

    private long calculatePrizeMoney(Map.Entry<Rank, Long> entry) {
        Rank rank = entry.getKey();
        long winCount = entry.getValue();
        return rank.multiple(winCount);
    }
```

메서드를 분리하여 의도를 분명히 드러내는 것은 어떨까요